packages/common/src/common/toast.tsx

```diff
@@ -0,0 +1,217 @@
+    right: var(--spacing-s);
+    width: var(--spacing-m);
+    height: var(--spacing-m);
+    background: black;
+    mask-image: url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='currentColor' fill-rule='evenodd' d='M18 7.5 13.5 12l4.5 4.5-1.5 1.5-4.5-4.5L7.5 18 6 16.5l4.5-4.5L6 7.5 7.5 6l4.5 4.5L16.5 6 18 7.5z' clip-rule='evenodd'/%3E%3C/svg%3E");
+  }
+  [class^="Toastify__progress-bar--wrp"] {
+    top: 0;
+    bottom: auto;
+    height: 8px;
+  }
+  [class^="Toastify__progress-bar--bg"] {
+    background: transparent;
+  }
+  [class^="Toastify__progress-bar "] {
+    opacity: 1;
+  }
+`;
+
+const HDSNotification = styled(Notification)`
+  && {
+    // We are using the border as progress bar, so we need to hide the default border
+    border-width: 0;
+  }
+  [class^="Notification-module_content"] {
+    padding-top: 8px;
+  }
+  .progress-bar {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 8px;
+    background-color: var(--notification-border-color);
+  }
+`;
+
+// any component can use this function to show a toast, as long as <ToastContainer /> is rendered somewhere in the app
+export default function toast({
+  label,
+  text,
+  type = "info",
+  duration,
+  dataTestId,
+  ariaLabel = label,
+  options,
+}: ToastProps): Id {
+  const toastOptions = options || {};
+  if (duration) {
+    toastOptions.autoClose = duration * 1000;
+  }
+  // toastify uses "warning" instead of "alert", so we need to convert it from HDS to toastify
+  toastOptions.type = type === "alert" ? "warning" : type;
+  switch (type) {
+    case "error":
+      toastOptions.icon = <IconErrorFill />;
+      break;
+    case "success":
+      toastOptions.icon = <IconCheckCircleFill />;
+      break;
+    default:
+      toastOptions.icon = <IconInfoCircleFill />;
+  }
+  return toastFn(
+    <ToastNotification
+      text={text}
+      type={type ?? "info"}
+      label={label}
+      ariaLabel={ariaLabel}
+      duration={duration && duration * 1000}
+      dataTestId={dataTestId}
+    />,
+    toastOptions
+  );
+}
+
+// This component is used to render the toast content, in our case it's a HDS Notification
+function ToastNotification({
+  type,
+  text,
+  label,
+  duration,
+  ariaLabel,
+  dataTestId,
+}: ToastProps): JSX.Element {
+  return (
+    <HDSNotification
+      type={type as NotificationType}
+      label={label}
+      notificationAriaLabel={ariaLabel ?? label ?? text}
+      dataTestId={dataTestId}
+    >
+      {
+        // toastify doesn't show the progress-bar if the toast doesn't have an autoClose duration defined
+        // so we need to add a custom "progress-bar" to comply with the HDS design
+        !duration && <div className="progress-bar" />
+      }
+      {text}
+    </HDSNotification>
+  );
+}
+
+// by default toastify uses "fixed" positioning,
+const AbsoluteToastContainer = styled(Container)`
+  position: absolute;
+`;
```

Comment:
What is wrong with `fixed`? if the user scrolls down two pages and gets a toast (some action success / fail), he's not gonna see it with `absolute`

Reply:
You're right.. in our use `fixed` probably works. When testing (showing all different toast types with and without labels) they couldn't fit the screen - and since it was `fixed` you couldn't reach the end of the list by scrolling. But we will (_hopefully_) never have that many toast - especially ones with a label, as those take way more vertical space than the ones without.